kernel/kernel-api/src/main/java/io/delta/kernel/CommitRangeBuilder.java

```diff
@@ -121,7 +121,6 @@ public static CommitBoundary atVersion(long version) {
      * @throws IllegalArgumentException if {@code timestamp} is negative
      */
     public static CommitBoundary atTimestamp(long timestamp, Snapshot latestSnapshot) {
-      checkArgument(timestamp >= 0, "Timestamp must be >= 0, but got: %d", timestamp);
```

Comment:
why this change?

Reply:
we use negative timestamps in a test or two -- I think technically this isn't an invalid timestamp (although we won't see it in any live tables) right?

Reply:
IMO we should keep this check (or just really explain to me why the negative timestamp is helpful for the test).

i.e. if a test has a negative timestmap --> great! it should hit this check and fail :) 

it's not really about what we should / probalby see in tables. it's an API. you can pass in a negative number --> and we should handle and validate that early 👍